app/src/main/java/org/piwigo/ui/login/LoginViewModel.java

```diff
@@ -108,13 +108,12 @@ void onLoginClick(@Nullable FABProgressCircle fabCircle) {
         if (fabCircle != null) {
             fabCircle.show();
         }
-        new URLHelper(newUrl -> {
-            testConnection(loginValid, newUrl);
-            url.set(newUrl);
-        }).execute(url.get());
+
+        testConnection(loginValid, URLHelper.INSTANCE.getUrlWithMethod(url.get(), "https"));
     }
 
-    void testConnection(boolean loginValid, String url){
+    void testConnection(boolean loginValid, String url) {
+        this.url.set(url);
```

Comment:
shouldn't we change the url only if we tested it successfully?

Reply:
Then url.get() would never contain "https" and we would never retry with HTTP..

Reply:
maybe yes. But it will cause some dirty flickering in the UI to first modify the text to add (or replace a user given http with) https and if there the connection fails change it back to http. Keep in mind, that `ObservableField<String> url` is directly coupled to the view...

It might work and on some (many) phones not being visible (if we have a good internet connection) but it's not really clean.

Decide on your own, whether you'd like to keep it.